subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecActionFactory.java

```diff
@@ -75,7 +75,8 @@ private DefaultExecActionFactory(
         this.javaModuleDetector = javaModuleDetector;
         this.buildCancellationToken = buildCancellationToken;
         this.executor = executor;
-        this.execHandleFactory = execHandleFactory;
+        // Use the same file resolver for low level process operations
+        this.execHandleFactory = execHandleFactory.withFileResolver(fileResolver);
```

Comment:
🤔 Is that the only usage that requires having a `fileResolver` from the project level?

What I am asking is: Is it possible that the regression impacts other features that were not yet reported.

Reply:
This regression should impact only ExecSpec related classes created through `DefaultExecActionFactory`, since we use a separate service for ExecHandle building now that could have incorrect `fileResolver` without this fix.

Other features could still use incorrect `fileResolver`, (see https://gradle.slack.com/archives/CC0HFTHRC/p1736850193806389?thread_ts=1736258726.623499&cid=CC0HFTHRC), but that was not introduced in 8.12 with this regression.

Reply:
FTR, which resolver is used where is in general a mystery to me that we should unravel. That said, I agree with @asodja that we should keep the scope to fixing the problems related to `exec` in this PR.

Reply:
Makes sense, was not asking to increase scope, just making sure there was not another regression (as opposed to an older bug) lurking around.